packages/skeleton-svelte/src/lib/components/Rating/Rating.svelte

```diff
@@ -1,116 +1,73 @@
 <script lang="ts">
+	import * as rating from '@zag-js/rating-group';
+	import { useMachine, normalizeProps } from '@zag-js/svelte';
+	import { useId } from '$lib/internal/use-id.js';
+	import { starEmpty, starHalf, starFull } from '$lib/internal/snippets.js';
 	import type { RatingProps } from './types.js';
 
+	// Props
 	let {
-		value = $bindable(0),
-		max = 5,
-		interactive = false,
-		step = 1,
+		value = $bindable(),
 		// Root
-		base = 'flex',
-		width = 'w-full',
-		justify = 'justify-center',
-		spaceX = 'space-x-2',
-		classes = '',
-		// Item ---
-		buttonBase = 'w-full h-full',
-		buttonPosition = 'relative',
-		buttonAspect = 'aspect-square',
-		buttonClasses = '',
-		// Icon Empty ---
-		emptyBase = 'absolute left-0 top-0 flex items-center justify-center',
-		emptyClip = '[clip-path:inset(0_0_0_var(--clipValue))] rtl:[clip-path:inset(0_var(--clipValue)_0_0)]',
-		emptyInteractive = 'size-full',
-		emptyStatic = 'w-fit',
-		emptyClasses = '',
-		// Icon Full ---
-		fullBase = 'absolute left-0 top-0 flex items-center justify-center',
-		fullClip = '[clip-path:inset(0_var(--clipValue)_0_0)] rtl:[clip-path:inset(0_0_0_var(--clipValue))]',
-		fullInteractive = 'size-full',
-		fullStatic = 'w-fit',
-		fullClasses = '',
-		// Events
-		onmousedown = () => {},
-		onkeydown = () => {},
-		// Snippets ---
-		iconEmpty,
-		iconFull
+		base = 'flex gap-1',
+		classes,
+		// Label
+		labelBase,
+		labelClasses,
+		// Item
+		itemBase = 'cursor-pointer data-[disabled]:cursor-not-allowed data-[readonly]:cursor-default',
```

Comment:
I might suggest reviewing how I handle `disabled` and `readonly` states styles for the Segment Control. I break them to individual style props so users can easily overwrite them. I also use the `.disabled` class which is a Skeleton-provided utility class.

There should be a visible difference between each state:

1. Default state - fully visible; fully interactive
2. Disabled state - half transparent; non-interactive
3. Read-Only state - fully visible; non-interactive

Reply:
That's a wondeful idea yeah, let me actually do that.